### PR TITLE
[octave] add new port

### PR DESCRIPTION
**Pull request by talregev**
- [x] Changes comply with the [maintainer guide](https://github.com/microsoft/vcpkg-docs/blob/main/vcpkg/contributing/maintainer-guide.md).
- [x] The name of the port matches an existing name for this component on https://repology.org/ if possible, and/or is strongly associated with that component on search engines.
- [x] Optional dependencies are resolved in exactly one way. For example, if the component is built with CMake, all `find_package` calls are REQUIRED, are satisfied by `vcpkg.json`'s declared dependencies, or disabled with [CMAKE_DISABLE_FIND_PACKAGE_Xxx](https://cmake.org/cmake/help/latest/variable/CMAKE_DISABLE_FIND_PACKAGE_PackageName.html).
- [x] The versioning scheme in `vcpkg.json` matches what upstream says.
- [x] The license declaration in `vcpkg.json` matches what upstream says.
- [x] The installed as the "copyright" file matches what upstream says.
- [x] The source code of the component installed comes from an authoritative source.
- [x] The generated "usage text" is accurate. See [adding-usage](https://github.com/microsoft/vcpkg-docs/blob/main/vcpkg/examples/adding-usage.md) for context.
- [x] The version database is fixed by rerunning `./vcpkg x-add-version --all` and committing the result.
- [x] Only one version is in the new port's versions file.
- [x] Only one version is added to each modified port's versions file.

### CODE REVIEW

File: ports/octave/vcpkg.json [hunk 1]
```diff
@@ -0,0 +1,27 @@
+{
+  "name": "octave",
+  "version": "9.1.0",
+  "description": "High-level interpreted language, primarily intended for numerical computations.",
+  "homepage": "https://octave.org/",
+  "license": "GPL-3.0-or-later",
+  "supports": "linux & !static",
```
Comment (dg0yt) [inline ports/octave/vcpkg.json:7]: This disables all vcpkg CI coverage.

File: ports/octave/vcpkg.json [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+  "dependencies": [
+    "blas",
+    "curl",
```
Comment (dg0yt) [inline ports/octave/vcpkg.json:10]: Almost always we want  `"default-features": false"` for this dependency. (Do you remember selecting the SSL backend?)
And maybe for some others, too.

**Review by dg0yt — CHANGES_REQUESTED** [review]
Just first feedback.
Too early to decide on spending more time on this PR.

File: ports/octave/vcpkg.json (hunk 1 above)
Comment (talregev) [reply ports/octave/vcpkg.json:7]: I know, but I only succeeded to compile this port with vcpkg on dynamic ubuntu.

**Comment by talregev** [comment]
> Just first feedback. Too early to decide on spending more time on this PR.

I admit that needed more work to compile this port on more triplets.

File: ports/octave/vcpkg.json (hunk 2 above)
Comment (talregev) [reply ports/octave/vcpkg.json]: done.

**Comment by MonicaLiu0311** [comment]
> > Just first feedback. Too early to decide on spending more time on this PR.
> 
> I admit that needed more work to compile this port on more triplets.

Note: I will be converting your PR to draft status. When you're ready, please revert to "ready for review".

File: ports/octave/vcpkg.json (hunk 1 above)
Comment (MonicaLiu0311) [reply ports/octave/vcpkg.json:7]: This is not allowed for the following reasons:
1. Currently, the official Linux we support is statically built. Your expression will cause our CI to be unable to detect and run. [triplets/x64-linux.cmake](https://github.com/microsoft/vcpkg/blob/master/triplets/x64-linux.cmake):
```
set(VCPKG_TARGET_ARCHITECTURE x64)
set(VCPKG_CRT_LINKAGE dynamic)
set(VCPKG_LIBRARY_LINKAGE static)

set(VCPKG_CMAKE_SYSTEM_NAME Linux)
```

2. I found that the upstream supports `windows & linux & osx`. In this case, `"support"` should be consistent with the upstream. You can write the failed triplets into [`ci.baselin.txt`](https://github.com/microsoft/vcpkg/blob/master/scripts/ci.baseline.txt):
image

**Comment by MonicaLiu0311** [comment]
Ping @talregev

**Comment by talregev** [comment]
@MonicaLiu0311 Thank you for your review.
I will take a look and I will respond

**Comment by talregev** [comment]
@MonicaLiu0311 
On the build system itself it say it cannot build with static:
https://github.com/gnu-octave/octave/blob/3fedbfc38b3f06f1ade3db4b0f011191a18b9f92/configure.ac#L637

**Comment by talregev** [comment]
> @MonicaLiu0311 On the build system itself it say it cannot build with static: https://github.com/gnu-octave/octave/blob/3fedbfc38b3f06f1ade3db4b0f011191a18b9f92/configure.ac#L637

I will try to patch and remove this error and see what will happen.

**Comment by dg0yt** [comment]
After reading the issue mentioned in configure.ac, it seems that this port os a candidate for
~~~
vcpkg_check_linkage(ONLY_DYNAMIC_LIBRARY)
~~~

**Comment by talregev** [comment]
Revert the port to be non static.
Wait all the ci that will fail, then I will add it to [ci.baselin.txt](https://github.com/microsoft/vcpkg/blob/master/scripts/ci.baseline.txt)

**Comment by MonicaLiu0311** [comment]
@talregev Is there any new progress?

**Comment by talregev** [comment]
for me, I finished develop this port.
It has only dynamic Linux working.
Should at least dynamic windows work?

If you want to review, go ahead, and let me know what you think.

**Comment by dg0yt** [comment]
There must be at least one CI configuration which builds the port.
And this is how it could be done: https://github.com/microsoft/vcpkg/pull/38243#issuecomment-2140605657.

**Comment by talregev** [comment]
@dg0yt Do you want to try this? You can use my commits.

**Comment by MonicaLiu0311** [comment]
Closing this PR since it seems that no progress is being made. Please ping us to reopen if work is still being done.

**Pull request closed without merging**